test/metabase/query_analysis/native_query_analyzer_test.clj

```diff
@@ -18,37 +19,91 @@
            ;; this is "Perv""e""rse"
            (#'nqa/field-query :f.name "\"Perv\"\"e\"\"rse\"")))))
 
-(deftest ^:parallel field-matching-test
-  (let [q (fn [sql]
-            (#'nqa/field-ids-for-native (mt/native-query {:query sql})))]
-    (testing "simple query matches"
-      (is (= {:explicit #{(mt/id :venues :id)} :implicit nil}
-             (q "select id from venues"))))
-    (testing "quotes stop case matching"
-      (is (= {:explicit nil :implicit nil}
-             (q "select \"id\" from venues")))
-      (is (= {:explicit #{(mt/id :venues :id)} :implicit nil}
-             (q "select \"ID\" from venues"))))
-    (testing "you can mix quoted and unquoted names"
-      (is (= {:explicit #{(mt/id :venues :id) (mt/id :venues :name)} :implicit nil}
-             (q "select v.\"ID\", v.name from venues v")))
-      (is (= {:explicit #{(mt/id :venues :id) (mt/id :venues :name)} :implicit nil}
-             (q "select v.`ID`, v.name from venues v"))))
-    (testing "It will find all relevant columns if query is not specific"
-      (is (= {:explicit #{(mt/id :venues :id) (mt/id :checkins :id)} :implicit nil}
-             (q "select id from venues join checkins"))))
-    (testing "But if you are specific - then it's a concrete field"
-      (is (= {:explicit #{(mt/id :venues :id)} :implicit nil}
-             (q "select v.id from venues v join checkins"))))
-    (testing "And wildcards are matching everything"
-      (is (= 10
-             (-> (q "select * from venues v join checkins")
-                 :implicit count)))
-      (is (= 6
-             (-> (q "select v.* from venues v join checkins")
-                 :implicit count))))
-
-    (when (not (contains? #{:snowflake :oracle} driver/*driver*))
-      (testing "Analysis does not fail due to keywords that are only reserved in other databases"
-        (is (= {:explicit #{(mt/id :venues :id)} :implicit nil}
-               (q "select id as final from venues")))))))
+(deftest ^:parallel consolidate-columns-test
+  (testing "We match references with known fields where possible, and remove redundancies"
+    (is (= [{:field-id 1, :table "t1", :column "c1"}
+            {:field-id 2, :table "t3", :column "c1"}
```

Comment:
Is it `t3.c1` (rather than `t1.c1`) because it's mentioned later on in list of columns?

Reply:
Ahh or because `group-by` returns both of them and then `mapcat` combines this stuff. Makes sense.